msal/src/Microsoft.Identity.Client/Logger.cs

```diff
@@ -127,17 +125,17 @@ public static LogCallback LogCallback
             {
                 lock (LockObj)
                 {
-                    if (_logCallback != null)
-                    {
-                        throw new ArgumentException("MSAL logging callback can only be set once per process and" +
-                                                   " should never change once set.");
-                    }
```

Comment:
@henrik-me  maybe this? https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/174

Reply:
That seems to be the reason... I don't get the reason. However we should naturally ensure that nothing breaks even if the logger is set to null. 

---
In reply to: [224878188](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1315#discussion_r224878188) [](ancestors = 224878188)

Reply:
The original bug (thanks jenny for the link) was basically "once its set, don't let someone else set it".  for testing, we need to be able to clear it.  so i can either leave it as it is (and then the original issue is present that someone could hijack the log callback, but technically that's a bug in your own app and we should technically allow multiple callbacks for that case) or I can make it so you can only set it back to null (which still allows a hijack {set null, set to my callback}).  

I'd prefer to leave it how it is even though it returns the ability to hijack since we should be able to safely clear the callback.

Reply:
I kind of agree. 

---
In reply to: [224880996](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1315#discussion_r224880996) [](ancestors = 224880996)